web/src/api/getState.ts

```diff
@@ -24,9 +24,26 @@ const getState = async (timeAverage: string): Promise<GridState> => {
 
 const useGetState = (): UseQueryResult<GridState> => {
   const [timeAverage] = useAtom(timeAverageAtom);
-  return useQuery<GridState>([QUERY_KEYS.STATE, { aggregate: timeAverage }], async () =>
-    getState(timeAverage)
+
+  // First fetch hour zero
+  const hour_zero = useQuery<GridState>(
+    [QUERY_KEYS.STATE, { aggregate: 'hour_zero' }],
+    async () => getState('hour_zero'),
+    {
+      enabled: timeAverage === 'hourly',
+    }
+  );
+
+  // Then fetch the rest of the data
+  const all_data = useQuery<GridState>(
+    [QUERY_KEYS.STATE, { aggregate: timeAverage }],
+    async () => getState(timeAverage),
+    {
+      // The query should not execute until the hour_zero query is done
+      enabled: Boolean(hour_zero.isLoading === false && hour_zero.data),
+    }
   );
```

Comment:
Something became evident in the preview deployment that might result in us having to change this.

Due to the fact that the backend PR hasn't been merged this always return a 400 status code.

While this would naturally resolve itself when we have merged said PR we should probably make it more robust in case the first query fails and still fetch the all data one since it includes the same data.

Another thing that came to mind is that we only need to fetch the first hour or hour zero when the user is first opening the app, not every time we fetch the hourly state (why didn't we think of this before? 😅).

I realize this adds complexity but I think it is worth it to make the app more robust and efficient.

Reply:
Good point on the 400, I'll attempt to make it proceed despite failing on the first one.

Since we now only fetch hourly once per hour, I don't think the additional request is a problem really :)